### DISCUSSION TITLE
How to bind a keymap to a key in normal state?

### DISCUSSION THREAD

**Discussion by witek** [meow-edit/meow#421] [Q&A]
I can use `(meow-normal-define-key "X" 'help)` to bind `X` to the `help` command when in meow normal mode. But how can I bind a key to a keymap instead?

I have a keymap defined in var `my-key-map`. Now I want this keymap to "start" when I press `X` when in meow normal mode. `(meow-normal-define-key "X" 'my-key-map)` does not work. It errors `Wrong type argument: commandp, 'my-key-map`.

**Comment by DogLooksGood** [comment]
Use `my-key-map`, instead of `'my-key-map`.

But it's better to do it with an alias:
```
(defalias 'my-key-map my-key-map)
(meow-normal-define-key "X" 'my-key-map)
```
This is how names of builtin keymaps handled.

**Reply by witek** [reply]
> (defalias 'my-key-map my-key-map)
> (meow-normal-define-key "X" 'my-key-map)

Evaluating this crashes with `(wrong-type-argument integer-or-marker-p nil)`

**Reply by DogLooksGood** [reply]
Sorry, it should be
```
(meow-normal-define-key '("X" . my-key-map))
```

**Reply by witek** [reply]
> Sorry, it should be
> 
> ```
> (meow-normal-define-key '("X" . my-key-map))
> ```

Evaluating this works. But when pressing `X` Emacs errors `Wrong type argument: commandp, (my-key-map)`.

**Reply by DogLooksGood** [reply]
I just tried this, it works
```
(defvar my-keymap (make-sparse-keymap))
(define-key my-keymap (kbd "a") 'other-window)
(defalias 'my-keymap my-keymap)
(meow-normal-define-key '("X" . my-keymap))
```

**Reply by witek** [reply]
You are right. I forgot `defalias`. Now it works. Thank you very much!
